app/models/course/assessment/answer.rb

```diff
@@ -225,4 +225,8 @@ def auto_grading_job_class(reduce_priority)
       Course::Assessment::Answer::AutoGradingJob
     end
   end
+
+  def auto_grading_time_limit
+    1.minutes
```

Comment:
How did you choose this?

Reply:
I'm just thinking of the usual evaluation time for normal programming question to be considered taking long time. This can also be any number, and I think 1 minute is unreasonably long already

Reply:
How can we be thorough/systematic about this?

Reply:
another way to get this correct (and which is quite expensive and might be over-engineer) is to let user input the threshold from frontend, and let it be nullable -> in which if it's null, we set up the default value as the auto grading time limit (just like how we did the `max_programming_time_limit`).

Reply:
No, as discussed, either do a random sample of how long answers take to grade, or make use of our default value for reference.

Why let the user input? If we do that then we run into the same problem of users wantonly setting high thresholds.

Reply:
done. I decided to use the `DEFAULT_TIMEOUT` defined within `Course::Assessment::ProgrammingEvaluationService` for the limit time